### ISSUE TITLE
suggest `map` or `map_err` when only the `Ok()` or `Err()` values are changed, respectively, in a pattern match

### ISSUE THREAD

**Issue by darklyspaced** [rust-lang/rust-clippy#11211]
### What it does

Checks for usage of a `match` of `if let` expression where only the `Ok()` **or** `Err()` values are modified; suggests the use of `map` or `map_err` instead.

### Advantage

 - Readability, usage of `Result::map` and `Result::map_err` is more concise

### Drawbacks

None that I can think of.

### Example

```rust
#[derive(PartialEq, Debug)]
struct Wrapper(i32);

let x = match fallible() {
    Ok(a) => Ok(Wrapper(a)),
    Err(err) => Err(err),
}

```

Could be written as:

```rust
# #[derive(PartialEq, Debug)]
# struct Wrapper(i32);

let x = falliable().map(|ok| Wrapper);
```

**Comment by mpalmer** [comment]
As a past master of the unnecessary `match` on `Result` (I've since gotten better), I definitely support the existence of this lint -- I'd like clippy to check my old code for cases where past-me was being a goose.  @darklyspaced, are you still interested in implementing this?  Otherwise, I could have a crack at it.

**Comment by darklyspaced** [comment]
yes, feel to have a go at it. i don't have much time nowadays unfortunately.

**Comment by samueltardieu** [comment]
Note that this is only true if the type of the `Err` variant is the same in both the match scrutinee and the match result.
